lib/src/local_working_copy.rs

```diff
@@ -116,10 +116,34 @@ use crate::working_copy::WorkingCopy;
 use crate::working_copy::WorkingCopyFactory;
 use crate::working_copy::WorkingCopyStateError;
 
+/// On-disk state of file executable bit.
+// TODO: maybe better to preserve the executable bit on all platforms, and
+// ignore conditionally? #3949
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct FileExecutableFlag(#[cfg(unix)] bool);
```

Comment:
I didn't realize you could `#[cfg]` inside tuple structs, but it makes sense. I'm going to keep that trick in my back pocket.